xamarin/Xamarin.Forms
# Fix #3798 - [Android] SeparatorColor of ListView is NOT updated dynamically

### Description of Change ###
The current issue (#3798 ) is not limited to `SeparatorColor` but also impacts `SeparatorVisibility` property (this also won't update dynamically).
Fixed this issue by improving the way we call `UpdateSeparatorVisibility` and `UpdateSeparatorColor` methods to draw the divider control by checking the current state of ListView Separator before calling it. For detailed description, please refer this comment: https://github.com/xamarin/Xamarin.Forms/issues/3798#issuecomment-541447547

### Issues Resolved ### 


- fixes #3798 
- It also fixes issue with `SeparatorVisibility` of ListView not updating dynamically

### API Changes ###

 
 None

### Platforms Affected ### 


- Android

### Behavioral/Visual Changes ###


If updating ListView's `SeparatorColor` or `SeparatorVisibility` dynamically in your app, you may see it working as expected.

### Before/After Screenshots ### 


**Before:**

![image](https://user-images.githubusercontent.com/17450941/67099088-e9d92f00-f1da-11e9-999e-12baf535e7fb.png)

**After:**

![image](https://user-images.githubusercontent.com/17450941/67099147-07a69400-f1db-11e9-810a-77f4c1a9b032.png)


### Testing Procedure ###


1. Create a `ListView` with few test items and set initial value of `SeparatorColor` property (e.g. Green)
2. Change the `SeparatorColor` property value (e.g. Red) dynamically from code (e.g. `ItemTapped` event). You should see the updated SeparatorColor (e.g. Red).
3. Repeat the above steps if you'd like to update `SeparatorVisibility` property dynamically.

- Or just run `Issue3798.xaml` from Control Gallery.

- I've also added a UI Test `Issue3798Test`.

### PR Checklist ###
